app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CommentsController < ApplicationController
+  before_action :set_repository
+  respond_to :js, :html
+
+  # POST /repositories/1/comments
+  # POST /repositories/1/comments.json
+  def create
+    @comment = @repository.comments.new(comment_params)
+    @comment.author = current_user
+    authorize @comment
+
+    if @comment.save
+      @repository.create_activity :create_comment,
+                                  owner:      current_user,
+                                  parameters: {
```

Comment:
Instead of doing this, just save: `comment_id`, `comment_create`, `repo_id` and `repo_name`. These are the attributes that are useful whenever the comment or the repo has been removed.